### PR TITLE
fix(appsflyer): event name casing in track payloads

### PR DESCRIPTION
**Pull request by saikumarrs**
## Description of the change

Updated the code to send the track event name without any casing modification to the AppsFlyer destination.

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

N/A

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/af/transform.js [hunk 1]
```diff
@@ -196,7 +195,7 @@ function processEventTypeTrack(message) {
       isMultiSupport
     );
   }
-  payload.eventName = eventName;
```
Comment (sanpj2292) [inline v0/destinations/af/transform.js:199]: Just curious, what would happen if `message.event` is not set while sending the payload ?

Comment (saikumarrs) [reply v0/destinations/af/transform.js:199]: I think it'll fail but it'd have been the scenario previously as well. 
Good point. So, we'll have to think through this while we build a more stable solution as part of this bug fix.
CC @utsabc

File: __tests__/data/af_router_output.json [hunk 2]
```diff
@@ -58,7 +58,7 @@
           "app_version_name": "1.0.0",
           "bundleIdentifier": "com.rudderlabs.javascript",
           "customer_user_id": "12345",
-          "eventValue": "{\"user_actual_role\":\"system_admin, system_user\",\"user_actual_id\":12345}",
+          "eventValue": "{\"properties\":{\"user_actual_role\":\"system_admin, system_user\",\"user_actual_id\":12345}}",
```
Comment (utsabc) [inline __tests__/data/af_router_output.json:61]: Properties is getting nested here is this expected?

File: v0/destinations/af/transform.js [hunk 3]
```diff
@@ -186,17 +185,12 @@ function processEventTypeTrack(message) {
       break;
     }
   }
-  let payload;
-  if (isUnIdentifiedEvent) {
```
Comment (saikumarrs) [inline v0/destinations/af/transform.js:190]: This variable is unused. Is that intentional? Let's clean up the other unused code if that's the case.

File: v0/destinations/af/transform.js (hunk 1 above)
Comment (saikumarrs) [reply v0/destinations/af/transform.js:199]: @yashasvibajpai can we look into this?

File: v0/destinations/af/transform.js [hunk 4]
```diff
@@ -185,16 +185,11 @@ function processEventTypeTrack(message) {
...
-    payload = getEventValueForUnIdentifiedTrackEvent(message);
-  } else {
-    payload = getEventValueMapFromMappingJson(
-      message,
-      mappingConfig[category.name],
-      isMultiSupport
-    );
-  }
+  const payload = getEventValueMapFromMappingJson(
```
Comment (utsabc) [inline v0/destinations/af/transform.js:188]: `isUnIdentifiedEvent` this not used anymore

**Review by utsabc — CHANGES_REQUESTED** [review]

File: __tests__/data/af_router_output.json (hunk 2 above)
Comment (utsabc) [reply __tests__/data/af_router_output.json:61]: Please make sure the existing test cases are not having any breaking changes

Comment (yashasvibajpai) [reply __tests__/data/af_router_output.json:61]: Yes, I changed this testcase only when I could see that earlier also properties was getting nested in `Order Completed` events. like [here](https://github.com/rudderlabs/rudder-transformer/blob/be5132d678576bbc9697492c78d494dd59ed3bf8/__tests__/data/af_output.json#L184)

File: v0/destinations/af/transform.js (hunk 1 above)
Comment (yashasvibajpai) [reply v0/destinations/af/transform.js:199]: Sure

File: v0/destinations/af/transform.js (hunk 3 above)
Comment (yashasvibajpai) [reply v0/destinations/af/transform.js:190]: Yes, it was useful as a check earlier, if the event was not belonging to one the declared categories of events with own mappings. Unused so removing it now.

File: v0/destinations/af/transform.js (hunk 4 above)
Comment (yashasvibajpai) [reply v0/destinations/af/transform.js:188]: Yes, it was useful as a check earlier, if the event was not belonging to one the declared categories of events with own mappings. Unused so removing it now.

File: v0/destinations/af/transform.js (hunk 1 above)
Comment (saikumarrs) [reply v0/destinations/af/transform.js:199]: CC @sanpj2292

File: v0/destinations/af/transform.js [hunk 5]
```diff
@@ -165,10 +165,8 @@ function processNonTrackEvents(message, eventName) {
 
 function processEventTypeTrack(message) {
   let isMultiSupport = true;
-  let isUnIdentifiedEvent = false;
   const evType = message.event && message.event.toLowerCase();
   let category = ConfigCategory.DEFAULT;
```
Comment (krishna2020) [inline v0/destinations/af/transform.js:169]: @utsabc have we analysed the impact of any existing customers with removal of lowercase conversion ?

**Review by utsabc — DISMISSED** [review]

File: v0/destinations/af/transform.js (hunk 5 above)
Comment (utsabc) [reply v0/destinations/af/transform.js:169]: We have analyzed the existing customers we are seeing none of them are sending. We have documented in the original card

**Review by utsabc — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

File: __tests__/data/af_router_output.json (hunk 2 above)
Comment (kishore-mega) [reply __tests__/data/af_router_output.json:61]: We are facing attribution issues in AppsFlyer because of this change.

Comment (utsabc) [reply __tests__/data/af_router_output.json:61]: Hello @kishore-mega we would be happy to help you out cab we understand what is the issue that you are facing were you having event names in Uppercase and mapped to attribution as lowercase?

Comment (kishore-mega) [reply __tests__/data/af_router_output.json:61]: Hi @utsabc ,
Let's continue the discussion regarding this issue [here](https://github.com/rudderlabs/rudder-transformer/issues/1451)

**Pull request merged**